README.md

```diff
@@ -66,25 +84,240 @@ npx cypress run --spec "cypress/e2e/journeys/example.spec.js" --config video=fal
+- [field content strings](https://github.com/UK-Export-Finance/exip/blob/main-application/src/ui/server/content-strings/fields/index.ts)
+
+Note: field IDs are in [constants](https://github.com/UK-Export-Finance/exip/blob/main-application/src/ui/server/constants) because an ID is not a content string.
+
+## DRY UI controllers and rendering
+
+Every page/controller requires the same content and links in order to meet design. For example, every page has a header, footer, service name and a back link. To save repetition, we've introduced **page variables functions** that should be used in every single controller.
+
+### Quote tool controllers
+
+Use [quoteCorePageVariables](https://github.com/UK-Export-Finance/exip/blob/main-application/src/ui/server/helpers/page-variables/core/quote/index.ts). Example usage:
+
+```js
+const get = (req: Request, res: Response) =>
+  res.render('template.njk', {
+    ...quoteCorePageVariables({ PAGE_CONTENT_STRINGS: PAGES.EXAMPLE, BACK_LINK: req.headers.referer }),
+    EXIT_REASON,
+  });
+```
+
+### Insurance controllers
+
+Use [insuranceCorePageVariables](https://github.com/UK-Export-Finance/exip/blob/main-application/src/ui/server/helpers/page-variables/core/insurance/index.ts). Example usage:
+
+```js
+const get = (req: Request, res: Response) =>
+  res.render('template.njk', {
+    ...insuranceCorePageVariables({ PAGE_CONTENT_STRINGS: PAGES.EXAMPLE, BACK_LINK: req.headers.referer }),
+    EXIT_REASON,
+  });
+```
+
+### Single input pages
+
+In the eligibility flows, there are lot of "yes or no" question pages with single radio button selection. The only difference between each of these pages is the field itself - e.g the title, id and hint. Therefore we also have a special page variables function, [singleInputPageVariables](https://github.com/UK-Export-Finance/exip/tree/main-application/src/ui/server/helpers/page-variables/single-input) that is exactly the same as the others, except that it can consume a field ID and automatically adds any hints or labels defined for the field ID. NBite Example usage:
+
+```js
+const get = (req: Request, res: Response) =>
+  res.render(
+    TEMPLATES.PROBLEM_WITH_SERVICE,
+    ...singleInputPageVariables({
+      {
+        FIELD_ID: 'fieldA'
+        PAGE_CONTENT_STRINGS: PAGES.EXAMPLE,
+      },
+      BACK_LINK: req.headers.referer,
+    }),
+  );
+```
+
+### Generic pages
+
+For controllers/pages that are not part of the Quote tool or Insurance, e.g "problem with this service", use [corePageVariables](https://github.com/UK-Export-Finance/exip/blob/main-application/src/ui/server/helpers/page-variables/core/index.ts). Example usage:
+
+```js
+const get = (req: Request, res: Response) =>
+  res.render(
+    TEMPLATES.PROBLEM_WITH_SERVICE,
+    corePageVariables({
+      PAGE_CONTENT_STRINGS: PAGES.EXAMPLE,
+      BACK_LINK: req.headers.referer,
+      PRODUCT: { DESCRIPTION: PRODUCT.DESCRIPTION.GENERIC },
```

Comment:
core page variables also has a START_ROUTE.  pedantic but can we add it here

Reply:
Good spot, added this